src/replication/replication_slave_node.cpp

```diff
@@ -265,12 +265,26 @@ namespace cubreplication
     stop_and_destroy_online_repl ();
   }
 
+  void slave_node::destroy_transfer_receiver ()
+  {
+    delete m_transfer_receiver;
+    m_transfer_receiver = NULL;
+  }
+
+  void slave_node::finish_fetch_from_stream ()
+  {
+    // this forces transefer_receiver to stream::commit_append all data it has received
+    destroy_transfer_receiver ();
```

Comment:
it is not obvious that destroy_transfer_receiver will also make sure to fetch all data from ex-master.

Is it not necessary to wait to fetch all data, even data that was not transferred yet?

Reply:
This only stops the transfer_receiver from appending to the stream and closes the receiving channel.
Only the following call, stream::wait_for_fetch_all is expected to do the job that you described. Do you think this should be made somehow clearer?

Reply:
Yes, please, I think it would help to have to trigger here a specific function that is clear to fetch all data from server.

> This only stops the transfer_receiver from appending to the stream and closes the receiving channel.
Correct me if I am wrong, but this is not expected final behavior. transfer_receiver should not be stopped from appending to the stream if it still has more data to receive from master. Consider the scenario of admin requesting a change of master. Should the slave that is promoted to master must get all the replication data from ex-master?

Reply:
if transfer channel is still working, finish_fetch_from_stream should wait for all data to be pushed.

Reply:
Regarding this, I've open an issue: http://jira.cubrid.org/browse/CBRD-23157